### ISSUE TITLE
NameError from table vstack

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#4208]
@taldcroft I'm getting this error:
```python
>>> from gammapy.data import EventListDataset
>>> filenames = ['hess_events_018406.fits.gz', 'run_0018406_std_eventlist.fits.gz']
>>> EventListDataset.vstack_from_files(filenames)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/deil/code/gammapy/gammapy/data/event_list.py", line 466, in vstack_from_files
    total_event_list = vstack_tables(event_lists, metadata_conflicts='silent')
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev13676-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 252, in vstack
    out = _vstack(tables, join_type, col_name_map)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev13676-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 734, in _vstack
    out_descrs = get_descrs(arrays, col_name_map)
  File "/Users/deil/Library/Python/3.4/lib/python/site-packages/astropy-1.1.dev13676-py3.4-macosx-10.10-x86_64.egg/astropy/table/operations.py", line 478, in get_descrs
    raise TableMergeError('Key columns {0!r} have different shape'.format(name))
NameError: name 'name' is not defined
```
Looks like a bug here:
```
$ pyflakes  astropy/table/operations.py 
astropy/table/operations.py:478: undefined name 'name'
```

**Comment by astrofrog** [comment]
@cdeil - is the bug in Astropy 1.0.4 or just master?

**Comment by cdeil** [comment]
It's in master. I don't know about 1.0.4.

**Comment by taldcroft** [comment]
Yeah, that should be `names` not `name`.  My pyflakes didn't catch it because in Py2.7 references leak out of list comprehensions.  And obviously the coverage needs improvement.

**Comment by embray** [comment]
https://github.com/astropy/astropy/blob/v1.0.x/astropy/table/operations.py#L480

**Comment by taldcroft** [comment]
Fix in #4213.
